core-metadata/src/main/java/org/apache/kylin/metadata/filter/FilterCodeSystemFactory.java

```diff
@@ -95,4 +99,32 @@ public int compare(Object o, Object t1) {
         }
     }
 
+    private static class DateCodeSystem implements IFilterCodeSystem<String> {
+
+        @Override
+        public boolean isNull(String code) {
+            return code == null;
+        }
+
+        @Override
+        public void serialize(String code, ByteBuffer buf) {
+            BytesUtil.writeUTFString(code, buf);
+        }
+
+        @Override
+        public String deserialize(ByteBuffer buf) {
+            return BytesUtil.readUTFString(buf);
+        }
+
+        @Override
+        public int compare(String o1, String o2) {
+            try {
+                long d1 = DateFormat.stringToMillis(o1);
+                long d2 = DateFormat.stringToMillis(o2);
+                return Long.compare(d1, d2);
+            } catch (IllegalArgumentException e) {
+                return o1.compareTo(o2);
```

Comment:
The exception shouldn't be caught silently; It should throw an error here, as using string comparator may return the wrong result.